plugin/pkg/admission/securitycontext/scdeny/admission.go

```diff
@@ -53,42 +53,53 @@ func (p *plugin) Admit(a admission.Attributes) (err error) {
-		return apierrors.NewForbidden(a.GetResource().GroupResource(), pod.Name, fmt.Errorf("SecurityContext.SupplementalGroups is forbidden"))
+	if err = checkPodSecurityContext(pod.Spec.SecurityContext); err != nil {
+		return apierrors.NewForbidden(a.GetResource().GroupResource(), pod.Name, err)
 	}
-	if pod.Spec.SecurityContext != nil {
-		if pod.Spec.SecurityContext.SELinuxOptions != nil {
-			return apierrors.NewForbidden(a.GetResource().GroupResource(), pod.Name, fmt.Errorf("pod.Spec.SecurityContext.SELinuxOptions is forbidden"))
-		}
-		if pod.Spec.SecurityContext.RunAsUser != nil {
-			return apierrors.NewForbidden(a.GetResource().GroupResource(), pod.Name, fmt.Errorf("pod.Spec.SecurityContext.RunAsUser is forbidden"))
+
+	for _, container := range pod.Spec.InitContainers {
+		if err = checkSecurityContext(container.SecurityContext); err != nil {
+			return apierrors.NewForbidden(a.GetResource().GroupResource(), pod.Name, err)
 		}
 	}
 
-	if pod.Spec.SecurityContext != nil && pod.Spec.SecurityContext.FSGroup != nil {
-		return apierrors.NewForbidden(a.GetResource().GroupResource(), pod.Name, fmt.Errorf("SecurityContext.FSGroup is forbidden"))
+	for _, container := range pod.Spec.Containers {
+		if err = checkSecurityContext(container.SecurityContext); err != nil {
+			return apierrors.NewForbidden(a.GetResource().GroupResource(), pod.Name, err)
+		}
 	}
 
-	for _, v := range pod.Spec.InitContainers {
-		if v.SecurityContext != nil {
-			if v.SecurityContext.SELinuxOptions != nil {
-				return apierrors.NewForbidden(a.GetResource().GroupResource(), pod.Name, fmt.Errorf("SecurityContext.SELinuxOptions is forbidden"))
-			}
-			if v.SecurityContext.RunAsUser != nil {
-				return apierrors.NewForbidden(a.GetResource().GroupResource(), pod.Name, fmt.Errorf("SecurityContext.RunAsUser is forbidden"))
-			}
+	return
+}
+
+func checkPodSecurityContext(podContext *api.PodSecurityContext) error {
+	if podContext != nil {
+		if podContext.SELinuxOptions != nil {
+			return fmt.Errorf("pod.Spec.SecurityContext.SELinuxOptions is forbidden")
+		}
+		if podContext.RunAsUser != nil {
+			return fmt.Errorf("pod.Spec.SecurityContext.RunAsUser is forbidden")
+		}
+		if podContext.SupplementalGroups != nil {
+			return fmt.Errorf("pod.Spec.SecurityContext.SupplementalGroups is forbidden")
+		}
+		if podContext.FSGroup != nil {
+			return fmt.Errorf("pod.Spec.SecurityContext.FSGroup is forbidden")
 		}
 	}
 
-	for _, v := range pod.Spec.Containers {
-		if v.SecurityContext != nil {
-			if v.SecurityContext.SELinuxOptions != nil {
-				return apierrors.NewForbidden(a.GetResource().GroupResource(), pod.Name, fmt.Errorf("SecurityContext.SELinuxOptions is forbidden"))
-			}
-			if v.SecurityContext.RunAsUser != nil {
-				return apierrors.NewForbidden(a.GetResource().GroupResource(), pod.Name, fmt.Errorf("SecurityContext.RunAsUser is forbidden"))
-			}
+	return nil
+}
+
+func checkSecurityContext(context *api.SecurityContext) error {
+	if context != nil {
```

Comment:
To reduce nesting level, please, use early returns. (See http://softwareengineering.stackexchange.com/questions/18454/should-i-return-from-a-function-early-or-use-an-if-statement)